tests/lib/rules/destructuring-assignment.js

```diff
@@ -265,8 +265,8 @@ ruleTester.run('destructuring-assignment', rule, {
     errors: [{message: 'Must use destructuring props assignment'}]
   }, {
     code: `
-      function hof() {
-        return (props) => <p>{props.a}</p>;
```

Comment:
let's find a way to keep this test case as-is, perhaps just by renaming `hof` to `HOF`?

Reply:
Thanks :+1:  Could you help figuring out what's wrong here ?
I reversed the changes as suggested but now is failing. It's suppose to throw an error message but it's not throwing.
Could be wrong, but I think that this unrelated test was failing for the wrong reason & my PR fix, just unmasked the bug.
**This appears to be a regular function that returns an anonymous React component.** I have research a lot and can't find an example of how that is used.
That's why previously thought it was a typo or bug.

Reply:
Wait ! You said something relevant in describing the original issue(#2842):
> The issue there isn’t that it’s nested - it’s that it’s treated as a component despite being used in an object literal with a key name that can’t be a custom component name.   
   
So if we take a look at the original issue(#2842) relevant code   
```
export function App() {
  return (
    <FormattedMessage
      values={{
        link:
          // The next line triggers react/no-multi-comp
          (text: string) => <a>{text}</a>,
      }}    /> ); }
```   

We see that is very similar to the [unrelated test](https://github.com/yannickcr/eslint-plugin-react/blob/edbbd7931902ec0b1cc1941df4155e39acc06f31/tests/lib/rules/destructuring-assignment.js#L267) that now is failing.   

So the questions is: Should this also be ignored ?:
```
      function hof() {
        return (props) => <p>{props.a}</p>;   
```
IF, they are similar & both should be ignored, then I should delete this code from [/tests/lib/rules/destructuring-assignment.js](https://github.com/yannickcr/eslint-plugin-react/blob/edbbd7931902ec0b1cc1941df4155e39acc06f31/tests/lib/rules/destructuring-assignment.js#L267) & add it to my valid code tests

Reply:
I think that `function hof() {` here should be ignored - however, if it's in a test case, it's likely an author error, and *should* be named "HOF" so it's a proper component that is *not* ignored.

Reply:
The only way I could find to keep this test was like this:
```
     function HOF(props) {
       return <p>{props.a}</p>;
     }
```
The reason, is that the way it was done before (returning an anonymous function), it was been ignored & the destructuring was not been checked. Remember that the purpose of this was testing invalid (not-destructuring) code but only on React components, not anonymous functions.

BTW, if you accept this, it needs to be rebased again. Since you rebased it, (& due to my inexperience), now it have a rebase conflict that I'm not able to resolve, sorry.